### PR TITLE
Make use of parity-codec "derive" crate feature

### PR DESCRIPTION
**Pull request by Robbepop**
Make use of `parity_codec`'s `derive` crate feature to simplify many imports.

### CODE REVIEW

**Comment by thiolliere** [comment]
use `./script/build.sh` to build Cargo.lock and wasm files, also CI has false positive you probably need to update node/runtime/src/lib.rs impl version of runtime for it not to complain

File: node/runtime/src/lib.rs [hunk 1]
```diff
@@ -61,7 +61,7 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
 	impl_name: create_runtime_str!("substrate-node"),
 	authoring_version: 10,
 	spec_version: 31,
-	impl_version: 32,
+	impl_version: 33,
```
Comment (bkchr) [inline node/runtime/src/lib.rs]: This is not required and not correct. You did not change the implementation. We can just ignore the CI false positive.

Comment (thiolliere) [reply node/runtime/src/lib.rs]: Oh I thought we couldn't, ok

Comment (bkchr) [reply node/runtime/src/lib.rs]: If you look at the CI, only `rust:stable` is tagged as `required`, all the other can be ignored. And `check:runtime` is producing a lot of false-positives, which is okay, but can be ignored if you are sure what you are doing :)

Comment (Robbepop) [reply node/runtime/src/lib.rs]: I ll simply revert it.

Comment (bkchr) [reply node/runtime/src/lib.rs]: Ty

Comment (Robbepop) [reply node/runtime/src/lib.rs]: done, ready for review

**Comment by DarkEld3r** [comment]
looks much less verbose. :+1: 
Is it still WIP?

**Comment by Robbepop** [comment]
> looks much less verbose. +1
> Is it still WIP?

Thanks!
No longer WIP, can be merged if approved.

**Pull request merged**